### ISSUE TITLE
`Alt-Ergo 2.5.2` doesn't compile with `Dune 3.13`

### ISSUE THREAD

**Issue by Halbaroth** [ocaml/dune#9944]
## Expected Behavior

Compilation of Alt-Ergo works with `dune 3.13` as it does with `dune 3.12`.

## Actual Behavior

If I run `opam install alt-ergo` with `Dune 3.13` installed in my opam switch, `dune` never ends the compilation of the package. 

## Reproduction

Run `opam install alt-ergo` with `Dune 3.13` or from the source of Alt-Ergo: 
```sh
git clone --depth 1 --branch v2.5.x https://github.com/OCamlPro/alt-ergo/
cd alt-ergo 
make dev-switch
eval $(opam env)
dune build @src/bin/text/all
``` 

## Specifications

- Version of `dune` (output of `dune --version`): `3.13.0`
- Version of `ocaml` (output of `ocamlc --version`): `5.1.1`
- Operating system (distribution and version): `Arch linux`
<details>
<summary>output of `opam list`</summary>
<pre>
base-bigarray       base
base-domains        base
base-nnp            base        Naked pointers prohibited in the OCaml heap
base-threads        base
base-unix           base
camlzip             1.11        Accessing compressed files in ZIP, GZIP and JAR format
cmdliner            1.2.0       Declarative definition of command line interfaces for OCaml
conf-gmp            4           Virtual package relying on a GMP lib system installation
conf-pkg-config     3           Check if pkg-config is installed and create an opam switch local pkgconfig folder
conf-zlib           1           Virtual package relying on zlib
csexp               1.5.2       Parsing and printing of S-expressions in Canonical form
dolmen              0.9         A parser library for automated deduction
dolmen_loop         0.9         A tool library for automated deduction tools
dolmen_type         0.9         A typechecker for automated deduction languages
dune                3.13.0      Fast, portable, and opinionated build system
dune-build-info     3.13.0      Embed build information inside executable
dune-private-libs   3.13.0      Private libraries of Dune
dune-site           3.13.0      Embed locations information inside executable and libraries
dyn                 3.13.0      Dynamic type
fmt                 0.9.0       OCaml Format pretty-printer combinators
gen                 1.1         Iterators for OCaml, both restartable and consumable
logs                0.7.0       Logging infrastructure for OCaml
menhir              20231231    An LR(1) parser generator
menhirCST           20231231    Runtime support library for parsers generated by Menhir
menhirLib           20231231    Runtime support library for parsers generated by Menhir
menhirSdk           20231231    Compile-time library for auxiliary tools related to Menhir
num                 1.5         The legacy Num library for arbitrary-precision integer and rational arithmetic
ocaml               5.1.1       The OCaml compiler (virtual package)
ocaml-compiler-libs v0.12.4     OCaml compiler libraries repackaged
ocaml-config        3           OCaml Switch Configuration
ocaml-system        5.1.1       The OCaml compiler (system version, from outside of opam)
ocamlbuild          0.14.3      OCamlbuild is a build system with builtin rules to easily build most OCaml projects
ocamlfind           1.9.6       A library manager for OCaml
ocplib-simplex      0.5         A library implementing a simplex algorithm, in a functional style, for solving systems of linear inequalities and optimizing linear objective functions
ordering            3.13.0      Element ordering
pp                  1.2.0       Pretty-printing library
pp_loc              2.1.0       Quote and highlight input fragments at a given source location
ppx_blob            0.7.2       Include a file as a string at compile time
ppx_derivers        1.2.1       Shared [@@deriving] plugin registry
ppxlib              0.31.0      Standard infrastructure for ppx rewriters
psmt2-frontend      0.4.0       The psmt2-frontend project
seq                 base        Compatibility package for OCaml's standard iterator type starting from 4.07.
sexplib0            v0.16.0     Library containing the definition of S-expressions and some base converters
spelll              0.4         Fuzzy string searching, using Levenshtein automaton
stdlib-shims        0.3.0       Backport some of the new stdlib features to older compiler
stdune              3.13.0      Dune's unstable standard library
topkg               1.0.7       The transitory OCaml software packager
uutf                1.0.3       Non-blocking streaming Unicode codec for OCaml
zarith              1.13        Implements arithmetic and logical operations over arbitrary-precision integers
</pre>
</details>

## Additional information 

I found the origin of the issue. In `src/bin/text/dune`, I have these rules:
```
(executable
  (name Main_text)
  (public_name alt-ergo)
  (package alt-ergo)
  (libraries alt_ergo_common)
  (link_flags (:include flags.dune))
  (modules Main_text)
  (promote (until-clean)))

; Rule to generate a man page for alt-ergo
(rule
  (target alt-ergo.1)
  (action (with-stdout-to %{target} (run alt-ergo --help=groff))))

; Install the man page
(install
 (files alt-ergo.1)
 (section man)
 (package alt-ergo))
```

If I replace the second rule by 
```
(rule
  (deps (:bin Main_text.exe))
  (target alt-ergo.1)
  (action (with-stdout-to %{target} (run %{bin} --help=groff))))
```

It works again :D 
I don't know if it's a bug of `Dune` or everything works by sheer luck before your last release.

**Comment by emillon** [comment]
hi,
I'm seeing this with 3.13.0 but this seems to be fixed with 3.13.1, so I think that this is related to #9769.
Do you confirm that upgrading to 3.13.1 works?

**Comment by emillon** [comment]
no problem!
